src/higley_lab_to_nwb/benisty_2024/interfaces/benisty_2024_cidansegmentation_interface.py

```diff
@@ -0,0 +1,131 @@
+
+        self.plane_segmentation_name = plane_segmentation_name
+        self.verbose = verbose
+
+    def get_metadata(self) -> DeepDict:
+        metadata = super().get_metadata()
+
+        # No need to update the metadata links for the default plane segmentation name
+        default_plane_segmentation_name = metadata["Ophys"]["ImageSegmentation"]["plane_segmentations"][0]["name"]
+        if self.plane_segmentation_name == default_plane_segmentation_name:
+            return metadata
+
+        plane_segmentation_metadata = metadata["Ophys"]["ImageSegmentation"]["plane_segmentations"][0]
+        default_plane_segmentation_name = plane_segmentation_metadata["name"]
+        default_plane_suffix = default_plane_segmentation_name.replace("PlaneSegmentation", "")
+        new_plane_name_suffix = self.plane_segmentation_name.replace("PlaneSegmentation", "")
+        plane_segmentation_metadata.update(
+            name=self.plane_segmentation_name,
+        )
+
+        fluorescence_metadata_per_plane = metadata["Ophys"]["Fluorescence"].pop(default_plane_segmentation_name)
+        # override the default name of the plane segmentation
+        metadata["Ophys"]["Fluorescence"][self.plane_segmentation_name] = fluorescence_metadata_per_plane
+        trace_names = [
+            property_name for property_name in fluorescence_metadata_per_plane.keys() if property_name != "name"
+        ]
+        comments=format_string_for_parameters_dict(dict_obj=self.segmentation_extractor.parameters_dict)
+        for trace_name in trace_names:
+            default_raw_traces_name = fluorescence_metadata_per_plane[trace_name]["name"].replace(
+                default_plane_suffix, ""
+            )
+            fluorescence_metadata_per_plane[trace_name].update(
+                name=default_raw_traces_name + new_plane_name_suffix,
+                comments=comments,
+            )
+
+        return metadata
+
+    def add_to_nwbfile(
+        self,
+        nwbfile: NWBFile,
+        metadata: Optional[dict] = None,
+        stub_test: bool = False,
+        stub_frames: int = 100,
+        include_roi_centroids: bool = True,
+        include_roi_acceptance: bool = False,
+        mask_type: Optional[str] = "image",  # Literal["image", "pixel", "voxel"]
+        plane_segmentation_name: Optional[str] = None,
+        iterator_options: Optional[dict] = None,
+        compression_options: Optional[dict] = None,
+    ):
+        super().add_to_nwbfile(
+            nwbfile=nwbfile,
+            metadata=metadata,
+            stub_test=stub_test,
+            stub_frames=stub_frames,
+            include_roi_centroids=include_roi_centroids,
+            include_roi_acceptance=include_roi_acceptance,
+            mask_type=mask_type,
+            plane_segmentation_name=self.plane_segmentation_name,
+            iterator_options=iterator_options,
+            compression_options=compression_options,
+        )
```

Comment:
Why do you need to override this method? If you just wanted to have control over `include_roi_acceptance` etc. you can just modify this in the `conversion_options`

Reply:
Right! Thanks